WalletWasabi.Tests/UnitTests/Hwi/HwiProcessBridgeMock.cs

```diff
@@ -241,7 +238,7 @@ private static bool CompareArguments(out bool isTestNet, string arguments, strin
 		private static bool CompareArguments(string arguments, string desired, bool useStartWith = false)
 			=> CompareArguments(out _, arguments, desired, useStartWith);
 
-		private static bool CompareGetXbpubArguments(string arguments, out string extPubKey)
+		private static bool CompareGetXbpubArguments(string arguments, out string? extPubKey)
```

Comment:
I don't know about these `?`s everywhere. If that's the idea of "fixing" the null magic warnings, then we may just turn it off...

Reply:
It correctly tells you what the out parameter can provide to you. I did not write that code, I don't understand it in depth and I don't want to touch too much code in one PR so whoever reviews this may flag this and say "hey, this returns `null` but it should not .." or whatever and it can be done later based on an issue.. but it must be constructive. Saying "turn it off" does not help me in any way.

Reply:
So why not let the warnings be there? So it correctly warns you that you are doing something wrong so "later" you are more likely to fix it?

Reply:
I don't understand your point of view here. `CompareGetXbpubArguments` behaves similar to `Dictionary.TryGetValue()`. It either returns `extPubKey` which is not-null or null. And that `?` correctly denotes it behaves that way. So I'm basically documenting current behavior. Also the author of the code designed the code so that `extPubKey` may be `null`. 

That `?` is documentation for everybody that it may return `null` so that when somebody uses this API he will be correctly warned if he does not check null accordingly. That makes sense to me.

Now, should it be `out string extPubKey` instead of `out string? extPubKey`? Well, that's good question. I don't know. But have I make the situation worse or better? I say I have improved it because now it's clear how it works. And anyone reading code that uses this method will know that it behaves like that. And he may fix it if it is not correct. If I leave it as it is. Then anyone reading code one layer above will think that the code cannot return null (which is not true) and may write incorrect code. 

Also I'm not sure why you guys are so against `null` values (I understand that without NRT enabled). When I see `null` used for "I don't have value here" I find it OK.

So what do we disagree about?

Reply:
When you see a null magic warning, then you have 3 choices:

1. leave it alone
2. hack around with a `?`
3. find out what's the conceptual issue here and fix it

1. is ok, 3. is great, 2. is bad. For example 3 would look like this in this specific case: [`389192b` (#4215)](https://github.com/zkSNACKs/WalletWasabi/pull/4215/commits/389192b90c200061215138d5b9f6cd952bb2226a)